### PR TITLE
modified getSOurceObjs to work with new ROPEs

### PR DESCRIPTION
**Pull request by michaelconnor468**

### CODE REVIEW

File: src/foam/nanos/rope/ROPEAuthorizer.java [hunk 1]
```diff
@@ -106,29 +106,43 @@ public boolean ropeSearch(ROPEActions operation, FObject obj, X x, String target
...
-          EQ(predicateProperty, (Long) retrieveProperty(obj, "get", "id"))
-        )
-        .select(new ArraySink()))
-        .getArray();
-
-      for ( FObject junctionObj : junctionObjs ) {
-        FObject sourceObj = rope.getIsInverse() ? (FObject) sourceDAO.find(((Long)retrieveProperty(junctionObj, "get", "targetId")).longValue()) : (FObject) sourceDAO.find(((Long)retrieveProperty(junctionObj, "get", "sourceId")).longValue());
-        sourceObjs.add(sourceObj);
-      }
+      foam.dao.ManyToManyRelationshipImpl relationship = (foam.dao.ManyToManyRelationshipImpl) retrieveProperty(obj, "get", rope.getRelationshipKey(), x);
```
Comment (noodlemoodle) [inline src/foam/nanos/rope/ROPEAuthorizer.java:109]: this returns a dao actually i think

**Review by noodlemoodle — CHANGES_REQUESTED** [review]

File: src/foam/nanos/rope/ROPEAuthorizer.java [hunk 2]
```diff
@@ -106,29 +106,28 @@ public boolean ropeSearch(ROPEActions operation, FObject obj, X x, String target
...
-      Object predicateProperty = rope.getIsInverse() ? rope.getJunctionModel().getAxiomByName("sourceId") : rope.getJunctionModel().getAxiomByName("targetId");
-      List<FObject> junctionObjs = ((ArraySink) junctionDAO
-        .where(
-          EQ(predicateProperty, (Long) retrieveProperty(obj, "get", "id"))
-        )
-        .select(new ArraySink()))
-        .getArray();
+      foam.dao.ManyToManyRelationshipImpl relationship = (foam.dao.ManyToManyRelationshipImpl) retrieveProperty(obj, "get", rope.getRelationshipKey(), x);
+      List<FObject> junctionObjs = (List<FObject>) ( (ArraySink) relationship.getJunctionDAO().where(
+        EQ(relationship.getJunction().SOURCE_ID, (Long) retrieveProperty(obj, "get", "id"))
```
Comment (noodlemoodle) [inline src/foam/nanos/rope/ROPEAuthorizer.java]: syntax error in this one in Mlang query 
refer to this for the first arg in EQ()
```Object predicateProperty = rope.getIsInverse() ? rope.getJunctionModel().getAxiomByName("sourceId") : rope.getJunctionModel().getAxiomByName("targetId");```

File: src/foam/nanos/rope/ROPEAuthorizer.java (hunk 1 above)
Comment (michaelconnor468) [reply src/foam/nanos/rope/ROPEAuthorizer.java:109]: i think not

File: src/foam/nanos/rope/ROPEAuthorizer.java (hunk 2 above)
Comment (michaelconnor468) [reply src/foam/nanos/rope/ROPEAuthorizer.java]: try again

**Pull request merged**